### ISSUE TITLE
Social cards - `&sup2;` not displayed correctly

### ISSUE THREAD

**Issue by rei-vilo** [squidfunk/mkdocs-material#7544]
### Context

One page has the title...

```
# Test of I&sup2;C bus".
```

...and is referenced in `mkdocs.yml` with 

```
        - Test of I&sup2;C bus: index.md
```

Both the page and the TOC render HTML code `&sup2;` correctly with `²` in `Test of I²C bus`.

### Bug description

However, the social card displays...

```
Test of I&sup2;C bus
```  

![index](https://github.com/user-attachments/assets/e3da7f67-2d63-4930-9597-b59be33f2c4f)

...instead of the expected 

```
Test of I²C bus
```  

* How to fix this? 
 
Thank you!

### Related links

- [Setting up social cards](https://squidfunk.github.io/mkdocs-material/setup/setting-up-social-cards/?h=social+card)
- Discussion https://github.com/squidfunk/mkdocs-material/discussions/7543

### Reproduction

Discussion https://github.com/squidfunk/mkdocs-material/discussions/7543

### Steps to reproduce

See attached minimal project [Mkdocs_bug.zip](https://github.com/user-attachments/files/17066331/Mkdocs_bug.zip)

### Browser

Firefox

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material/), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/) using the [built-in info plugin](https://squidfunk.github.io/mkdocs-material/plugins/info/).

**Comment by squidfunk** [comment]
Thanks for reporting. Fixed in b655e0780.

**Comment by rei-vilo** [comment]
That was fast, thank you very much!

**Comment by squidfunk** [comment]
Released as part of [9.5.36](https://squidfunk.github.io/mkdocs-material/changelog/#9.5.36).

**Comment by rei-vilo** [comment]
![2cf4b0876d3b23483d93764b34d563ba](https://github.com/user-attachments/assets/ae4e948b-c549-4c1b-8bcb-5280a03275a3)

Thank you!
